### ISSUE TITLE
The experiment cannot be reproduced.

### ISSUE THREAD

**Issue by hitwhssj** [gemcollector/RL-ViGen#7]
When training svea or pieg on the walker_walk task of the dmc environment, even if I set the seed, the results are still different each time. why is that? Looking forward to your reply.

**Comment by hitwhssj** [comment]
> When training svea or pieg on the walker_walk task of the dmc environment, even if I set the seed, the results are still different each time. why is that? Looking forward to your reply.

@gemcollector 👀

**Comment by gemcollector** [comment]
Hi there, this is inherit randomness in each run (even if seed is fixed). This comes from the fact that rendering is non deterministic. You can read more about it here: https://github.com/google-deepmind/dm_control/issues/100.

Therefore, we use 5 random seeds and report average performance to provide more reliable and stable metric.

**Comment by hitwhssj** [comment]
Thank you for your reply.
Actually, I found a problem besides the dmc environment. When using "svea.py" to train on walker walk, after some steps, I found that the input parameter "obs" is same, but the "obs" which comes from "self.encoder(obs.unsqueeze(0))" seems to be different even the seed is same. Are there any problems？Does this illustrate the uncertainty caused by "self.encoder"?
```
    def act(self, obs, step, eval_mode):
        obs = torch.as_tensor(obs, device=self.device)
        obs = self.encoder(obs.unsqueeze(0))
        # set same seed and run several times, the "obs" there seems to be different after some steps
        stddev = utils.schedule(self.stddev_schedule, step)
        dist = self.actor(obs, stddev)
        if eval_mode:
            action = dist.mean
        else:
            action = dist.sample(clip=None)
            if step < self.num_expl_steps:
                action.uniform_(-1.0, 1.0)
        return action.cpu().numpy()[0]
```

**Comment by gemcollector** [comment]
The algorithm part of our repo is basically based on DrQ-v2. You can check this repo for further information. 

There are might be some other reasons:

1. After some training steps, the encoder will be changed. So if you pass the same input through the encoder, you cannot get the same results.

2. Some torch relevant seed might be fixed. But I am not sure whether it will influence the performance.

If you have further questions about the experiment setting, you can contact me via email :)

**Comment by hitwhssj** [comment]
> The algorithm part of our repo is basically based on DrQ-v2. You can check this repo for further information.
> 
> There are might be some other reasons:
> 
> 1. After some training steps, the encoder will be changed. So if you pass the same input through the encoder, you cannot get the same results.
> 2. Some torch relevant seed might be fixed. But I am not sure whether it will influence the performance.
> 
> If you have further questions about the experiment setting, you can contact me via email :)

Got it, thanks for the reply.
